src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/SessionParameterValueResolverTest.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Tests\Controller\ArgumentResolver;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
+use Symfony\Component\HttpKernel\Attribute\MapSessionParameter;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\SessionParameterValueResolver;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+class SessionParameterValueResolverTest extends TestCase
+{
+    private ValueResolverInterface $resolver;
+
+    private Request $request;
+
+    protected function setUp(): void
+    {
+        $this->resolver = new SessionParameterValueResolver();
+
+        $session = new Session(new MockArraySessionStorage());
+        $this->request = Request::create('/');
+        $this->request->setSession($session);
+    }
+
+    public function testSkipWhenNoAttribute()
+    {
+        $metadata = new ArgumentMetadata('browsingContext', 'string', false, true, false);
+
+        $this->assertSame([], $this->resolver->resolve($this->request, $metadata));
+    }
+
+    public function testSkipWhenNoSession()
+    {
+        $metadata = new ArgumentMetadata('MySessionObject', BasicSessionParameter::class, false, false, false, attributes: [new MapSessionParameter()]);
+
+        $this->assertSame([], $this->resolver->resolve(Request::create('/'), $metadata));
+    }
+
+    /**
+     * @dataProvider invalidArgumentTypeProvider
```

Comment:
please put the implementation of each data provider just after the methods that use them